packages/astro/src/core/config/schema.ts

```diff
@@ -366,22 +366,14 @@ export function createRelativeSchema(cmd: string, fileProtocolRoot: string) {
 		) {
 			config.build.client = new URL('./dist/client/', config.outDir);
 		}
-		const trimmedBase = trimSlashes(config.base);
 
-		// If there is no base but there is a base for site config, warn.
-		const sitePathname = config.site && new URL(config.site).pathname;
-		if (!trimmedBase.length && sitePathname && sitePathname !== '/') {
-			config.base = sitePathname;
-			/* eslint-disable no-console */
-			console.warn(`The site configuration value includes a pathname of ${sitePathname} but there is no base configuration.
-
-A future version of Astro will stop using the site pathname when producing <link> and <script> tags. Set your site's base with the base configuration.`);
-		}
```

Comment:
It looks like this had never worked for some time now. We do re-assign `config.base = sitePathname`, but the logic after this re-assign `config.base` based on something else again. So this compat feature had already been removed.